tests/std/tests/GH_000685_condition_variable_any/test.cpp

```diff
@@ -79,9 +82,136 @@ namespace {
         assert(m.num_locks() == 4);
 #endif // _HAS_CXX20
     }
+
+    // Minimal example inspired by LWG-4301, modified due to missing latch in C++11 to 20
+    // and generalized to test all overloads of condition_variable{_any}::wait_{for, until}
+    // Idea: Make the main thread wait for a CV with a short timeout and modify it from another thread in the meantime.
+    // If the main thread wait times out after short time, the modification did not influence the ongoing wait
+    template <typename CV>
+    void test_timeout_immutable(int test_number, int retries_remaining = 5) {
+        printf("\ntest %d\n", test_number);
+
+        mutex m;
+        CV cv;
+        unique_lock<mutex> lock(m); // Prevent other thread from modifying timeout too early
+
+        // Start with very short timeout and let other_thread change it to very large while main thread is waiting
+        constexpr auto short_timeout = 1s;
+        constexpr auto long_timeout  = 10s;
+
+        atomic_flag waiting_for_other_thread = ATOMIC_FLAG_INIT;

```

Comment:
This macro is weird (not exported by `import std;`!) and unnecessary.

WG21-N5014 \[atomics.flag\]/21: "The macro `ATOMIC_FLAG_INIT` is defined in such a way that it can be used to initialize an object of type `atomic_flag` to the clear state."

We can just value-init, see /4: "`constexpr atomic_flag::atomic_flag() noexcept;` *Effects:* Initializes `*this` to the clear state."